extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/repositories/repositories-module.js

```diff
@@ -0,0 +1,172 @@
+;
+(function() {
+    'use strict';
+
+    angular.module('PaperUI.services.repositories', [ 'PaperUI.services.rest' ]).factory('Repository', Repository);
+
+    function Repository() {
+        return {
+            create : RepositoryImpl
+        }
+    }
+
+    function RepositoryImpl($q, $rootScope, remoteService, dataType, staticData, getOneFunction, idParameterName, elmentId) {
+        var self = this;
+
+        this.cacheEnabled = true;
+        this.dirty = false;
+        this.initialFetch = false;
+        this.staticData = staticData
+        this.singleElements = getOneFunction ? {} : null;
+
+        return {
+            add : add,
+            remove : remove,
+            update : update,
+            getAll : getAll,
+            getOne : getOne,
+            find : find,
+            filter : filter,
+            findByIndex : findByIndex,
+            setDirty : setDirty
+        };
+
+        function setDirty() {
+            self.dirty = true;
+        }
+
+        function getAll(callback, refresh) {
+            if (typeof callback === 'boolean') {
+                refresh = true;
```

Comment:
for the `getAll(false)` case `refresh` should be assigned with the `callback` value